src/obj-ignore.c

```diff
@@ -660,15 +663,15 @@ void ignore_drop(struct player *p)
 				if (!verify_object("Really take off and drop", obj, p)) {
 					/* Hack - inscribe the item with !d to prevent repeated
 					 * confirmations. */
-					const char *inscription = quark_str(obj->note);
+					const char *inscription = obj->note;
 
 					if (inscription == NULL) {
-						obj->note = quark_add("!d");
+						obj->note = string_make("!d");
 					} else {
 						char buffer[1024];
 						my_strcpy(buffer, inscription, sizeof(buffer));
 						my_strcat(buffer, "!d", sizeof(buffer));
-						obj->note = quark_add(buffer);
+						obj->note = string_make(buffer);
```

Comment:
Release the preexisting obj->note before this assignment to avoid a memory leak.